src/cpp/simplehttpserver.cpp

```diff
@@ -553,7 +553,11 @@ SimpleHttpRequest *SimpleHttpServer::takeNext()
 	if(!d->pending.isEmpty())
 	{
 		SimpleHttpRequest *req = d->pending.takeFirst();
-		req->disconnect(d);
+		auto conn = d->finishedConnections.find(req);
+		if (conn != d->finishedConnections.end()) {
+			conn->second.disconnect();
```

Comment:
Since these are scoped connections, maybe deleting the entry is enough and this call is not needed?

Reply:
very true! done